Orange/canvas/application/canvasmain.py

```diff
@@ -108,75 +109,38 @@ def paintEvent(self, event):
         pass
 
 
-class DockableWindow(QDockWidget):
-    def __init__(self, *args, **kwargs):
-        QDockWidget.__init__(self, *args, **kwargs)
-
-        # Fist show after floating
-        self.__firstShow = True
-        # Flags to use while floating
-        self.__windowFlags = Qt.Window
-        self.setWindowFlags(self.__windowFlags)
-        self.topLevelChanged.connect(self.__on_topLevelChanged)
-        self.visibilityChanged.connect(self.__on_visbilityChanged)
-
-        self.__closeAction = QAction(self.tr("Close"), self,
-                                     shortcut=QKeySequence.Close,
-                                     triggered=self.close,
-                                     enabled=self.isFloating())
-        self.topLevelChanged.connect(self.__closeAction.setEnabled)
-        self.addAction(self.__closeAction)
-
-    def setFloatingWindowFlags(self, flags):
-        """
-        Set `windowFlags` to use while the widget is floating (undocked).
-        """
-        if self.__windowFlags != flags:
-            self.__windowFlags = flags
-            if self.isFloating():
-                self.__fixWindowFlags()
-
-    def paintEvent(self, event):
-        # HACK: Preventing calls to QDockWidget.paintEvent() seems to fix the
-        #
-        #     QPainter::begin: Paint device returned engine == 0, type: 3
-        #     QPainter...
-        #
-        # wall-of-text error that filled the buffers of our consoles.
-        #
-        # Without understanding reasons or implications, but with
-        # no immediate apparent negative side effects:
-        pass
-
-    def floatingWindowFlags(self):
-        """
-        Return the `windowFlags` used when the widget is floating.
-        """
-        return self.__windowFlags
+try:
+    QKeySequence.Cancel
+except AttributeError:  # < Qt 5.?
+    QKeySequence.Cancel = QKeySequence(Qt.Key_Escape)
 
-    def __fixWindowFlags(self):
-        if self.isFloating():
-            update_window_flags(self, self.__windowFlags)
-
-    def __on_topLevelChanged(self, floating):
-        if floating:
-            self.__firstShow = True
-            self.__fixWindowFlags()
-
-    def __on_visbilityChanged(self, visible):
-        if visible and self.isFloating() and self.__firstShow:
-            self.__firstShow = False
-            self.__fixWindowFlags()
 
+class DockWidget(QDockWidget):
```

Comment:
That's correct. Docks that can be docked or undocked but they aren't windows in themselves.